### PR TITLE
[TASK] Improve documentation for confvals

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 1]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+        :required: true
         :type: string or LLL reference
-        :Scope: Display
-        :Path: $GLOBALS > TCA > [table] > columns > [field]
 
         The name of the field as shown in the form.
 
-Configuration value with default value
---------------------------------------
+Example: Configuration value with default value adn custom parameter
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:56]: `and`

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 2]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
 with different contexts you can define individual configuration schemas for
 each context.
 
+..  contents:: Table of contents
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:30]: I would prefer moving examples after each confval directive or behind all directives, but not in front. You have inserted a very detailed example for fast copy&paste in CheatSheet.

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 3]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
 ========
 
 Required configuration value
 ----------------------------
 
 ..  confval:: label
-
-    :Required: true
+    :name: some-unique-label
+    :required: true
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:42]: It does not make sense which value you set. Set it to `0`, `false`, Sphinx native `False` this directive will always show `true`. Don't know what's wrong here. May you have a look?
Else: If `required` is not mandatory, the `required` directive has to be removed completely.

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 4]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+
+`:type:`
+    The type of the configuration value.
+`:default:`
+    The default value
+`:required:`
+    Is the configuration value required.
+`:name:`
+    The unique identifier, reserved internally by reStructuredText.
+`:class:`
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:99]: I havn't found `class` here: https://github.com/sphinx-toolbox/sphinx-toolbox/blob/master/sphinx_toolbox/confval.py. Do we have a special implementation for TYPO3 docs?

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 5]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+
+    ..  confval-menu::
+        :display: table
+        :type:
+        :default:
+        :exclude: confval-1, confval-2
+
+
+If you use a confval menu together with nested confvals it will only list
+its child confvals. This is usefull if you have several groups of confvals on
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:149]: `useful`

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 6]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+the same page and want to list them in separate menues:
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:150]: `menus`

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 7]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+    :type:
+
+    ..  confval:: confval-1
+        :type: string
+
+        Some Description
+
+    ..  confval:: confval-2
+        :type: int
+        :default: 'Hello World'
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:163]: Sure, it's just an example. But on type `int` I would prefer an int value as default.

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 8]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+
+            Some Description
+
+Confval-menu directive API
+==========================
+
+The confval-menu directive has the following options:
+
+`:display:`
+    'table', 'list', 'tree': Different display forms, just try them out
```
Comment (froemken) [inline Documentation/WritingReST/Reference/Code/Confval.rst:190]: OK for me, but why not using the ticks instead of single quotes to highlight them

**Review by froemken — CHANGES_REQUESTED** [review]

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/WritingReST/Reference/Code/Confval.rst:99]: class and name are defined and reserved for any directive.

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 5 above)
Comment (linawolf) [inline Documentation/WritingReST/Reference/Code/Confval.rst:149]:
```suggestion
its child confvals. This is useful if you have several groups of confvals on
```

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 9]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+    ..  confval-menu::
+        :display: table
+        :type:
+        :default:
+        :exclude: confval-1, confval-2
+
+
+If you use a confval menu together with nested confvals it will only list
+its child confvals. This is useful if you have several groups of confvals on
+the same page and want to list them in separate menues:
```
Comment (linawolf) [inline Documentation/WritingReST/Reference/Code/Confval.rst:150]:
```suggestion
the same page and want to list them in separate menus:
```

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 7 above)
Comment (linawolf) [reply Documentation/WritingReST/Reference/Code/Confval.rst:163]: woops, ack

File: Documentation/WritingReST/Reference/Code/Confval.rst [hunk 10]
```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
...
+
+            Some Description
+
+Confval-menu directive API
+==========================
+
+The confval-menu directive has the following options:
+
+`:display:`
+    'table', 'list', 'tree': Different display forms, just try them out
```
Comment (linawolf) [inline Documentation/WritingReST/Reference/Code/Confval.rst:190]:
```suggestion
    `table`, `list`, `tree`: Different display forms, just try them out
```

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 1 above)
Comment (linawolf) [inline Documentation/WritingReST/Reference/Code/Confval.rst:56]:
```suggestion
Example: Configuration value with default value and custom parameter
```

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/WritingReST/Reference/Code/Confval.rst:42]: I will investigate

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/WritingReST/Reference/Code/Confval.rst:30]: It seems like it is easier for newbies to have the examples first. And with the contents menu the experienced user can jump to the api right away

File: Documentation/WritingReST/Reference/Code/Confval.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/WritingReST/Reference/Code/Confval.rst:42]: If it is so I would say its a bug and I will solve it in the render-guides

**Comment by linawolf** [comment]
Thanks for your review

**Review by froemken — APPROVED** [review]
Thank you for the update and your explanations

**Pull request merged**